code/lib/react-dom-shim/package.json

```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@storybook/react-dom-shim",
+  "version": "7.0.0-beta.53",
+  "description": "",
+  "keywords": [
+    "storybook"
+  ],
+  "homepage": "https://github.com/storybookjs/storybook/tree/main/lib/react-dom-shim",
+  "bugs": {
+    "url": "https://github.com/storybookjs/storybook/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/storybookjs/storybook.git",
+    "directory": "lib/react-dom-shim"
+  },
+  "funding": {
+    "type": "opencollective",
+    "url": "https://opencollective.com/storybook"
+  },
+  "license": "MIT",
+  "sideEffects": false,
+  "exports": {
+    ".": {
+      "node": "./dist/index.js",
+      "require": "./dist/index.js",
+      "import": "./dist/index.mjs",
+      "types": "./dist/index.d.ts"
+    },
+    "./react-18": {
+      "node": "./dist/react-18.js",
+      "require": "./dist/react-18.js",
+      "import": "./dist/react-18.mjs",
+      "types": "./dist/react-18.d.ts"
+    },
+    "./dist/preset": {
```

Comment:
Shouldn't this just be `./preset`?

Reply:
I was cargo culting this, so maybe @ndelangen can tell me what I should do here.

Reply:
I think having `./dist/preset` is the right thing to do: https://github.com/storybookjs/storybook/pull/21197/files#r1115437543

We do not need to use auto-loading of preview/manager entries here, since they do not exist.

Reply:
I think the rule is the export names should match to the files they are exporting (in order to support tools that don't support export maps), with one exception: `'.'` which can be anything as long as it mirrors the `"module"` field.

Reply:
Ok, but then what is the reason for ./preset.js at the root?  I thought that we had that file for tools that don't support export maps, and for those that do support it we can use the map to simplify the import to just `/preset`

Reply:
There should be no presets.js file at the root, in this case, because the place where we inject the addon, we can be precise.

The issue is that user's (by our own design) tend to be imprecise, and just specify:
`@storybook/addon-foo`

Here:
https://github.com/storybookjs/storybook/blob/0bd3c1fce640e1c24f6307b40a150f52867448cb/code/lib/core-common/src/presets.ts#L119-L124

...we convert that, and try a bunch of patterns.. This code uses `require.resolve`, and `resolve-from`, which do support export maps, but always resolve to `require` or `node` "fields".

The complicating factor is that users ALSO (again by our own design) tend to import the preview runtime via the top-level-export:
```ts
import { foo } from '@storybook/addon-foo';
```

This tends to also support export maps, but not always, it's dependent on the builder.

But just to throw in some extra salt into the mix: some code-analysis tools, don't support export maps fully yet, and might complain about `@storybook/addon-foo` nor resolving to a file, therefore we have to ensure the top-level-export is the preview runtime code; and it should be listed as the `main`-field, and the `module`-field in package.json.

Sorry this is so overly complex.

I'm hoping that after 7.0 we might move to ESM only (for node too) and clean up the export maps.

---

I think as a general rule, the auto loading of `manager`, `preview`, `preset` files is something we should treat as a ease-of-use feature for users, and if we can be precise, we should be precise.
I actually think we should do a bit of an overhaul of the essentials addon in this regards, because I don't think it's setting quite the right standard for "how to have 1 addon add another addon".

I hope this helps @IanVS 🙏